src/Features/CSharp/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,149 @@
+                // Example: 
+                // string[] args;
+                // String.Format("{0}{1}{2}", args);
+                return IsArgumentListNotPassingArrayToParams(
+                    invocation.ArgumentList.Arguments[1].Expression,
+                    invocationSymbol,
+                    formatMethods,
+                    semanticModel,
+                    cancellationToken);
+            }
+
+            return false;
+        }
+
+        protected override bool IsStringLiteral(LiteralExpressionSyntax firstArgument) =>
+            firstArgument?.Token.IsKind(SyntaxKind.StringLiteralToken) == true;
+
+        protected override InterpolatedStringExpressionSyntax VisitArguments(
+            ImmutableArray<ExpressionSyntax> expandedArguments,
+            InterpolatedStringExpressionSyntax interpolatedString)
+        {
+            return interpolatedString.ReplaceNodes(interpolatedString.Contents, (oldNode, newNode) =>
+            {
+                var node = newNode as InterpolationSyntax;
+                if (node == null)
+                {
+                    return newNode;
+                }
+
+                var literalExpression = node.Expression as LiteralExpressionSyntax;
+                if (literalExpression != null && literalExpression.IsKind(SyntaxKind.NumericLiteralExpression))
+                {
+                    var index = (int)literalExpression.Token.Value;
+                    if (index >= 0 && index < expandedArguments.Length)
+                    {
+                        return node.WithExpression(FixTrivia(expandedArguments[index]));
+                    }
+                }
+
+                return newNode;
+            });
+        }
+
+        /// <summary>
+        /// Since C# interpolations cannot be on more than one line, we need to remove newlines if possible.
+        /// </summary>
+        public static ExpressionSyntax FixTrivia(ExpressionSyntax node) =>
+            node.ReplaceTokens(node.DescendantTokens(descendIntoTrivia: true), RemoveTriviaForTokens);
+
+        private static SyntaxToken RemoveTriviaForTokens(SyntaxToken originalToken, SyntaxToken rewrittenToken) =>
+            rewrittenToken
+                .WithLeadingTrivia(
+                    ReplaceSyntaxKindsWithElasticSpace(rewrittenToken.LeadingTrivia, SyntaxKind.WhitespaceTrivia, SyntaxKind.EndOfLineTrivia))
+                .WithTrailingTrivia(
+                    ReplaceSyntaxKindsWithElasticSpace(rewrittenToken.TrailingTrivia, SyntaxKind.WhitespaceTrivia, SyntaxKind.EndOfLineTrivia));
+
+        private static SyntaxTriviaList ReplaceSyntaxKindsWithElasticSpace(SyntaxTriviaList trivialList, params SyntaxKind[] kinds) =>
+            trivialList.Select(x => kinds.Any(y => x.IsKind(y)) ? SyntaxFactory.ElasticSpace : x).ToSyntaxTriviaList();
+
+        private static ExpressionSyntax CastAndParenthesize(ExpressionSyntax expression, SemanticModel semanticModel) =>
```

Comment:
See Cast

---
In reply to: [65440808](https://github.com/dotnet/roslyn/pull/11415#discussion_r65440808) [](ancestors = 65440808)